philosophy.md

```diff
@@ -1,12 +1,36 @@
 
-## Don't assume anything.
+WP-CLI's guiding principle is to be the fastest and canonical way to manage WordPress:
 
-Bundled commands MUST work on any given WordPress install (provided that it's a new enough version), no matter how it's configured and no matter how much data it has.
+* Speed is inherently present in the simplicity of the command-line interface, and the ability to chain multiple commands together into a more complex operation.
+* When functionality ends up in WP-CLI, it should be good enough to become the defacto solution to the problem.
+* "Managing WordPress" is the ever-evolving problem space in which we operate. WP-CLI should always focus on making *existing management patterns* more efficient.
 
-As a corollary, bundled commands should not assume the presence of any plugin or theme.
+This page contains a list of guidelines that should inform decisions related to scope, command organization and behavior:
+
+## Good commands are simple, interact with WordPress, and solve one unique problem.
+
+When a command implements existing WordPress functionality (e.g. `wp export`), it should mirror and re-use existing WordPress behavior as much as possible. Doing so ensures to meet the user's assumed expectations of behavior.
+
+When a command adds functionality not present in the WordPress admin (e.g. `wp search-replace`), it should solve one unique problem and interact with WordPress in some way. Good commands are born of true user need; for a new proposed WP-CLI command, there probably should be existing prior art in the ecosystem.
+
+If a command provides useful functionality, but it doesn't have anything to do with WordPress, it doesn't belong in WP-CLI. Similarly, an idea for a command is often better implemented as documentation.
+
+## Consistency makes an enjoyable user experience.
+
+Whenever possible, commands should emulate one another with consistent usage behavior, argument names, and documentation.
+
+## Decisions, not options.
+
+WordPress' philosophy [states](https://wordpress.org/about/philosophy/):
+
+> When making decisions these are the users we consider first. A great example of this consideration is software options. Every time you give a user an option, you are asking them to make a decision. When a user doesn’t care or understand the option this ultimately leads to frustration. As developers we sometimes feel that providing options for everything is a good thing, you can never have too many choices, right? Ultimately these choices end up being technical ones, choices that the average end user has no interest in. It’s our duty as developers to make smart design decisions and avoid putting the weight of technical choices on our end users.
```

Comment:
This sentence reads badly because it is missing a prior subject/object:
```
When making decisions these are the users we consider first.
```
Maybe something like this reads better:
```
When making decisions [it is the users] we consider first.
```

Reply:
1d1415d